libraries/chain/block_state.cpp

```diff
@@ -298,6 +298,14 @@ void block_state::verify_qc(const valid_quorum_certificate& qc) const {
                invalid_qc_claim, "signature validation failed" );
 }
 
+qc_claim_t block_state::extract_qc_claim() const {
+   auto itr = header_exts.lower_bound(instant_finality_extension::extension_id());
+   if (itr == header_exts.end())
+      return {};
+   const auto& if_ext = std::get<instant_finality_extension>(itr->second);
+   return if_ext.qc_claim;
+}
```

Comment:
We really should have a generic function to extract an extension (like `template<typename Ext> Ext extract_extension()`, but checking if the extension is present and returning a std::optional).